### PR TITLE
Ms editing

### LINKED ISSUE #8: Editing A Note
When viewing an existing reminder, the user can click on an “Edit” button where they can modify the title, notes, or date of the reminder.

* When editing a note, there is a “Save” button that commits the changes to the server.
* When the user hits the save button, they return to viewing that individual note.

Things to keep in mind:
* What should your route look like to edit a reminder?
* Next Steps: What are similarities between the new and edit routes that we can refactor?

### PR DESCRIPTION
**Pull request by maiastone**
## Purpose

When viewing an existing reminder, the user can click on an “Edit” button where they can modify the title, notes, or date of the reminder.   
When editing a note, there is a “Save” button that commits the changes to the server.   
When the user hits the save button, they return to viewing that individual note.   

## Approach

In this case, an if else block statement was used in the reminder.hbs file.  When the user clicks edit, the action toggles the property isEditing.  When ifEditing is applied, the viewer sees input fields, rather than just the html markup.  When edits are complete, the viewer can hit save.  The action applied to the save finds the corresponding, active, record from the store by id, sets the new properties, and saves the active reminder.  Then, the property isEditing, is toggled back, so the viewer sees the original html markup, and the input fields are no longer visible.

### Learning

This was primarily solved by using the Ember docs, with a few helpful examples found while browsing.   
[Ember docs on controllers with an example for isExpanded](https://guides.emberjs.com/v2.8.0/controllers/)   
[Ember docs on input helpers and textareas](https://guides.emberjs.com/v2.8.0/templates/input-helpers/)   
[Ember docs on creating, updating, and deleting records](https://guides.emberjs.com/v2.8.0/models/creating-updating-and-deleting-records/)

### Test coverage 

Acceptance tests were added confirming the ability to edit title, date, and notes.   

### Screenshots

new screenshot 😉 

Closes #8    
@nfosterky @martensonbj @Tman22

### CODE REVIEW

**Comment by maiastone** [comment]
Enjoy the new screenshot

**Comment by maiastone** [comment]
Editing screenshot above and this PR has no merge conflicts.  
@Tman22 @martensonbj @nfosterky

File: app/templates/reminders/reminder.hbs [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
-  <h3 class='spec-reminder-date'>{{model.date}}</h3>
-  {{#if model.notes}}
-    <h3 class='spec-reminder-notes'>{{model.notes}}</h3>
+
+  {{#if isEditing}}
+
+  <h2>Title{{input type='text' class='spec-edit-title' value=model.title}}</h2>
+  <h3>Notes{{textarea class='spec-edit-notes' value=model.notes}}</h3>
+  <h3>Date{{input type='date' class='spec-edit-date' value=model.date}}</h3>
+  <button class='save' {{action 'saveReminder' model on 'submit'}}>Save</button>
```
Comment (nfosterky) [inline app/templates/reminders/reminder.hbs]: Indent everything in the if statement

File: app/routes/reminder.js [hunk 2]
```diff
@@ -3,5 +3,5 @@ import Ember from 'ember';
 export default Ember.Route.extend({
   model(params) {
     return this.get('store').find('reminder', params.reminder_id);
-  }
+  },
```
Comment (Tman22) [inline app/routes/reminder.js]: Take out comma please :)

Comment (maiastone) [inline app/routes/reminder.js]: I'd love to!

File: app/templates/reminders/reminder.hbs (hunk 1 above)
Comment (maiastone) [inline app/templates/reminders/reminder.hbs]: Done, thanks.

**Pull request merged**